librad/src/profile.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright © 2019-2021 The Radicle Foundation <hello@radicle.foundation>
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use std::{
+    env,
+    fs,
+    io,
+    path,
+    path::{Path, PathBuf},
+};
+use uuid::Uuid;
+
+use crate::paths::{project_dirs, Paths};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Invalid profile ID in RAD_PROFILE environment variable: {id}")]
```

Comment:
The convention for error messages is starting with lowercase, as mentioned in `radicle-decisions`